sync/class.jetpack-sync-module-themes.php

```diff
@@ -28,22 +28,24 @@ public function init_listeners( $callable ) {
 		add_action( 'jetpack_cleared_inactive_widgets', $callable );
 		add_action( 'jetpack_widget_reordered', $callable, 10, 2 );
 		add_filter( 'widget_update_callback', array( $this, 'sync_widget_edit' ), 10, 4 );
-		add_action( 'jetpack_widget_edited', $callable );
+		add_action( 'jetpack_widget_edited', $callable, 10, 2 );
 	}
 
 	public function sync_widget_edit( $instance, $new_instance, $old_instance, $widget_object ) {
 		$widget = array(
 			'name' => $widget_object->name,
 			'id' => $widget_object->id,
 		);
+		$is_new = empty( $old_instance ) ? true : false;
+
 		/**
 		 * Trigger action to alert $callable sync listener that a widget was edited
 		 *
 		 * @since 5.0.0
 		 *
 		 * @param string $widget_name, Name of edited widget
 		 */
-		do_action( 'jetpack_widget_edited', $widget );
+		do_action( 'jetpack_widget_edited', $widget, $is_new );
```

Comment:
I don’t think so. `jetpack_widget_added` is kicked off by listening for `update_option_sidebars_widgets` and firing other actions in turn, https://github.com/Automattic/jetpack/blob/a96a1d58eea4c73f0484d76859e10cd03e5fe3f0/sync/class.jetpack-sync-module-themes.php#L346. I didn’t write any of the relevant code and am unfamiliar with the “gotchas” that (presumably) required it to be written this way. In my opinion, refactoring the way Jetpack works here (arguably unnecessarily, since it seems to be working) wouldn’t be an optimal use of time right now in light of the need to finish E2E testing. Of course, I’m happy to look further into it, regardless, if you’d like me to. Please let me know